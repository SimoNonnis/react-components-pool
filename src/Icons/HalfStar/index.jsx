import React from 'react';
import PropTypes from 'prop-types';

/**
 * HalfStar Icon.
 * @param {number} width
 * @param {number} height
 * @param {string} fill
 * @param {string} className
 * @param {string} classNamePath
 * @returns {React.Component}
 */
function HalfStar({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 100 100"
      width={width}
      height={height}
      className={className}
    >
      <title>HalfStar Icon</title>
      <path
        className={classNamePath}
        fill={fill}
        d="M100,38.76,62.3,38.1,50,2.44,37.7,38.1,0,38.76,30.12,61.48l-11,36.06L50,75.93l30.9,21.63-11-36.08ZM74.24,88.39,50,71.43h0V13.78h0l9.64,28,29.58.52L65.61,60.1Z"
      />
    </svg>
  );
}

HalfStar.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

HalfStar.defaultProps = {
  width: 40,
  height: 40,
  fill: 'currentColor',
  className: 'halfStarIcon',
  classNamePath: 'emptyStarIconPath',
};

export default HalfStar;
