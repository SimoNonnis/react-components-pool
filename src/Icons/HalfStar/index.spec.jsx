/* eslint-env jest */
import React from 'react';

import HalfStar from './';

describe('Test <HalfStar /> icon', () => {
  it('should render HalfStar icon correctly with default props', () => {
    const wrapper = shallow(<HalfStar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render HalfStar icon correctly with props', () => {
    const wrapper = shallow(<HalfStar fill="#8A2BE2" height={30} width={30} />);
    expect(wrapper).toMatchSnapshot();
  });
});
