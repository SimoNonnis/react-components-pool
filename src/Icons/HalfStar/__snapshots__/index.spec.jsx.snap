// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <HalfStar /> icon should render HalfStar icon correctly with default props 1`] = `
<svg
  className="halfStarIcon"
  height={40}
  viewBox="0 0 100 100"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    HalfStar Icon
  </title>
  <path
    className="emptyStarIconPath"
    d="M100,38.76,62.3,38.1,50,2.44,37.7,38.1,0,38.76,30.12,61.48l-11,36.06L50,75.93l30.9,21.63-11-36.08ZM74.24,88.39,50,71.43h0V13.78h0l9.64,28,29.58.52L65.61,60.1Z"
    fill="currentColor"
  />
</svg>
`;

exports[`Test <HalfStar /> icon should render HalfStar icon correctly with props 1`] = `
<svg
  className="halfStarIcon"
  height={30}
  viewBox="0 0 100 100"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    HalfStar Icon
  </title>
  <path
    className="emptyStarIconPath"
    d="M100,38.76,62.3,38.1,50,2.44,37.7,38.1,0,38.76,30.12,61.48l-11,36.06L50,75.93l30.9,21.63-11-36.08ZM74.24,88.39,50,71.43h0V13.78h0l9.64,28,29.58.52L65.61,60.1Z"
    fill="#8A2BE2"
  />
</svg>
`;
